app/models/host/managed.rb

```diff
@@ -742,7 +747,7 @@ def smart_proxies
 
   def smart_proxy_ids
     ids = []
-    [subnet, hostgroup.try(:subnet)].compact.each do |s|
+    [subnet, subnet6, hostgroup.try(:subnet), hostgroup.try(:subnet6)].compact.each do |s|
```

Comment:
Nitpick: I'd perhaps do IPv4 first (host, hostgroup), then IPv6 for consistency.